dali/kernels/tensor_view.h

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_TENSOR_VIEW_H_
+#define DALI_KERNELS_TENSOR_VIEW_H_
+
+#include "shape.h"
+
+namespace tensor {
+
+struct EmptyBackendTag {};
```

Comment:
Not an issue, but a question: I used an enumeration whereas you use a tag type. What is the logic behind that?
Was it extensibility (you can add a type without modifying the enum defined in the API) or something else?

Reply:
I was thinking that having a type allows us to put some additional info in there. Still, with enums we can use some traits, that is probably a matter of preference.